### PR TITLE
Topic recognition

### PR DESCRIPTION
**Pull request by mhansen444**
To whom it may concern,

This pull request contains my implementation of the Improved U-Net segmentation of the ISIC2018 dataset for the COMP3710 report.
At this time it achieves 0.79 dice accuracy on the test dataset.
Included in this pull request are:
dataset.py, modules.py, train.py, predict.py and test_driver.py as advised by the spec. README.MD is also accompanied by some images.
For more information please see the README.

Thanks,
Michael Jorgen Hansen
s4428306

### CODE REVIEW

**Comment by wangzhaomxy** [comment]
This is an initial inspection, no action is required at this point.

- Well organized files.
- Algorithm solves the problem appropriately. Average Dice Coefficient in testing dataset: 0.79. Predicted Mask output looks good.
- Correctly using TensorFlow to construct the Improved U-Net models and functions.
- Good coding design.
- Sufficient docstring and comment in the code, and good references.
- Easy difficulty problem, no higher difficulty marks.

**Comment by shakes76** [comment]
# Marking
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate design and implementation
Good spacing and comments
Header blocks
## Recognition Problem 
Solves problem 
Driver Script present
File structure present
Shows Usage & Demo & Visualisation & Data usage
Module present
Commenting
No Data leakage
Difficulty: Easy -10
## Commit Log
Meaningful commit messages
Progressive commits used
## Documentation
ReadMe acceptable
Model/technical explanation
Good Description and Comments 
Markdown used and PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
No Feedback required
Request Description good
